rllib/algorithms/bc/bc_rl_module.py

```diff
@@ -0,0 +1,53 @@
+import abc
+from typing import Any, Type
+
+from ray.rllib.core.models.specs.typing import SpecType
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import ExperimentalAPI, override
+
+
+@ExperimentalAPI
+class BCRLModule(RLModule, abc.ABC):
+    @override(RLModule)
+    def setup(self):
+        # __sphinx_doc_begin__
+        catalog = self.config.get_catalog()
+
+        # Build models from catalog
+        self.encoder = catalog.build_encoder(framework=self.framework)
+        self.pi = catalog.build_pi_head(framework=self.framework)
+
+        self.action_dist_cls = catalog.get_action_dist_cls(framework=self.framework)
+
+    @override(RLModule)
+    def get_train_action_dist_cls(self) -> Type[Distribution]:
+        return self.action_dist_cls
+
+    @override(RLModule)
+    def get_exploration_action_dist_cls(self) -> Type[Distribution]:
+        return self.action_dist_cls
+
+    @override(RLModule)
+    def get_inference_action_dist_cls(self) -> Type[Distribution]:
+        return self.action_dist_cls
+
+    @override(RLModule)
+    def get_initial_state(self) -> Any:
```

Comment:
I took this from the `rl_module.py`:

```python
@OverrideToImplementCustomLogic
def get_initial_state(self) -> Any:
    """Returns the initial state of the module.

    This can be used for recurrent models.
    """
    return {}
```
But when I look into the `core/models/tf/encoder.py` I find indeed:

```python
@override(Model)
def get_initial_state(self):
    return {
        "h": tf.zeros((self.config.num_layers, self.config.hidden_dim)),
    }
```
As we use here the default encoder I think it is better to follow your suggestion and define the return type to `Union[dict, List[TensorType]]`.